dev/AppNotifications/AppNotificationBuilder/AppNotificationBuilder.cpp

```diff
@@ -237,134 +237,150 @@ namespace winrt::Microsoft::Windows::AppNotifications::Builder::implementation
         m_useButtonStyle = value.ButtonStyle() != AppNotificationButtonStyle::Default;
 
         return *this;
     }
 
-    winrt::Microsoft::Windows::AppNotifications::Builder::AppNotificationBuilder AppNotificationBuilder::SetTag(winrt::hstring const& value)
+    winrt::Microsoft::Windows::AppNotifications::Builder::AppNotificationBuilder AppNotificationBuilder::SetTag(hstring const& value)
     {
         m_tag = value;
         return *this;
     }
 
-    winrt::Microsoft::Windows::AppNotifications::Builder::AppNotificationBuilder AppNotificationBuilder::SetGroup(winrt::hstring const& value)
+    winrt::Microsoft::Windows::AppNotifications::Builder::AppNotificationBuilder AppNotificationBuilder::SetGroup(hstring const& value)
     {
         m_group = value;
         return *this;
     }
 
-    winrt::Microsoft::Windows::AppNotifications::AppNotification AppNotificationBuilder::BuildNotification()
+    std::wstring AppNotificationBuilder::GetDuration()
     {
-        winrt::hstring xmlResult{ L"<toast" };
-
-        // Add duration attribute if set
-        if (m_duration != AppNotificationDuration::Default)
-        {
-            xmlResult = xmlResult + L" duration=\"long\"";
-        }
+        return m_duration == AppNotificationDuration::Default ? L"" : L" duration='long'";
+    }
 
+    std::wstring AppNotificationBuilder::GetScenario()
+    {
         // Add scenario attribute if set
-        if (m_scenario != AppNotificationScenario::Default)
+        switch (m_scenario)
         {
-            winrt::hstring scenario{};
-            switch (m_scenario)
-            {
-            case AppNotificationScenario::Alarm:
-                scenario = L"\"alarm\"";
-                break;
-            case AppNotificationScenario::Reminder:
-                scenario = L"\"reminder\"";
-                break;
-            case AppNotificationScenario::IncomingCall:
-                scenario = L"\"incomingCall\"";
-                break;
-            case AppNotificationScenario::Urgent:
-                scenario = L"\"urgent\"";
-                break;
-            }
-
-            xmlResult = xmlResult + L" scenario=" + scenario;
+        case AppNotificationScenario::Alarm:
+            return L" scenario='alarm'";
+        case AppNotificationScenario::Reminder:
+            return L" scenario='reminder'";
+        case AppNotificationScenario::IncomingCall:
+            return L" scenario='incomingCall'";
+        case AppNotificationScenario::Urgent:
+            return L" scenario='urgent'";
+        default:
+            return {};
         }
+    }
 
+    std::wstring AppNotificationBuilder::GetArguments()
+    {
         // Add launch arguments if given arguments
         if (m_arguments.Size())
         {
-            xmlResult = xmlResult + L" launch=\"";
-
-            std::wstring arguments{};
+            std::wstring arguments{ };
```

Comment:
NIT: remove space `std::wstring argments{};`

Or even remove the `{}`. For void ctor `std::wstring arguments;` suffices.